doc/classes/Vector3.xml

```diff
@@ -465,6 +465,17 @@
 				Returns a vector with each component set to one or negative one, depending on the signs of this vector's components, or zero if the component is zero, by calling [method @GlobalScope.sign] on each component.
 			</description>
 		</method>
+		<method name="signed_angle_to">
+			<return type="float">
+			</return>
+			<argument index="0" name="to" type="Vector3">
+			</argument>
+			<argument index="1" name="axis" type="Vector3">
+			</argument>
+			<description>
+				Returns signed angle to the given vector, in radians. Sign of the angle is positive for clockwise rotation and negative for anti-clockwise rotation around reference [code]axis[/code].
```

Comment:
Godot uses a right-handed coordinate system for 3D, which means that we follow the right-hand rule. Rotations are positive in a counter-clockwise direction, not a clockwise one.

![](https://i.imgur.com/0vYoTQH.png)

Reply:
But angle_to in Vector2 gives positive angle for clockwise direction and negative for anti-clockwise direction...

Reply:
2D and 3D are different coordinate systems - the 2D coordinate system in Godot is not equivalent to the XY plane of the 3D coordinate system. In 2D Y points downwards, so with the right-hand rule positive angles are then clockwise.

Reply:
Describing things in terms of left-hand and right-hand isn't exactly relevant for 2D because there is no Z axis to spin around. If we treat Z as pointing out of the screen (like what happens with Godot's Z index), then 2D follows the left-hand rule. If we treat Z as pointing into the screen, then 2D follows the right-hand rule (it would be like if we took Godot's 3D coordinates and spun them around the X axis by 180 degrees).

Reply:
I'm sorry then. I aware about right-hand rule in vector math, but I want it to be consistent with 2D.

My mistake. I will remember about that in the future. Thank you for pointing that out.